fds/services/dvc_service.py

```diff
@@ -253,12 +254,37 @@ def commit(self, auto_confirm: bool) -> Any:
         execute_command(commit_cmd, capture_output=False)
 
     @staticmethod
-    def push(remote: str) -> Any:
+    def __execute_push_command(remote: str) -> CompletedProcess:
         push_cmd = ["dvc", "push"]
+        # We will only retry once
         if remote:
             push_cmd.append("-r")
             push_cmd.append(remote)
-        execute_command(push_cmd, capture_output=False)
+        return execute_command(
+            push_cmd,
+            capture_output=False,
+            capture_output_and_write_to_stdout=True
+        )
+
+    def push(self, remote: str) -> Any:
+        push_output_bytes = self.__execute_push_command(remote)
+        # Check if its unauthorized
+        if '401 Unauthorized' in convert_bytes_to_string(push_output_bytes.stderr):
```

Comment:
This will still ask for username and password even if they have already been configured and the user is just not authorized.
The more correct way to do it (maybe in some future version) is to check if a username and password is already defined, and ask or do something different if so.